File: sample_code/Sample.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.ResultSet;
import java.sql.Statement;

public class Sample {
    public void getUser(String username) {
        try {
            // hardcoded credentials
            Connection conn = DriverManager.getConnection("jdbc:mysql://localhost/test", "root", "password");
            Statement stmt = conn.createStatement();
            
            // security vulnerability: sql injection
            String query = "SELECT * FROM users WHERE username = '" + username + "'";
            ResultSet rs = stmt.executeQuery(query);
            
            while(rs.next()) {
                System.out.println(rs.getString("username"));
            }
        } catch (Exception e) {
            // bad practice: weak exception handling
            e.printStackTrace();
        }
    }
}

File: sample_code/sample.js
const express = require('express');
const app = express();

app.get('/user', (req, res) => {
    // security vulnerability: sql injection
    const userId = req.query.id;
    const query = "SELECT * FROM users WHERE id = " + userId;
    
    // bad practice: console.log in production code
    console.log("Query:", query);
    
    res.send("Executed query: " + query);
});

app.listen(3000, () => console.log('Server running'));
